acme/tests/client_test.py

```diff
@@ -93,6 +101,10 @@ class BackwardsCompatibleClientV2Test(ClientTestBase):
 
     def setUp(self):
         super().setUp()
+
+        # For some reason, required to suppress warnings on mock.patch('acme.client.Client')
+        warnings.filterwarnings('ignore', '.* in acme.client', DeprecationWarning)
```

Comment:
What's the purpose of the changes in this file? `acme/acme/client.py` isn't modified in this PR and we're ignoring this warning in `pytest.ini`.

With that said, if we/you want to make this change for some reason, we probably can, but I don't love this line of code here as I believe it repeatedly makes changes to the global warnings filter every time a `BackwardsCompatibleClientV2Test` test is run and that these changes persist for the rest of the test run even outside this class & module. Can we avoid making persistent changes? If not, what if we did this in the top level of the module so it is executed only once when the module is loaded?

Reply:
I don't know. For some reason, each of the `mock.patch('acme.client.Client')` calls trigger the warning, despite a module filter and `pytest.ini`, unless I put the filter in `setUp` or the tests themselves.  

>If not, what if we did this in the top level of the module so it is executed only once when the module is loaded?

Tried, didn't seem to work. 

I'm going to do the `__enter__`/`__exit__` thing in `setUp`/`tearDown`. It's test code and we're getting rid of it hopefully very soon.

Reply:
Yep, that's what I was going for with the change.